packages/pubsub/src/subscription.js

```diff
@@ -765,76 +856,105 @@ Subscription.prototype.pull = function(options, callback) {
  *
  * subscription.seek(date, callback);
  */
-Subscription.prototype.seek = function(snapshot, callback) {
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'seek'
-  };
+Subscription.prototype.seek = function(snapshot, gaxOpts, callback) {
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
   var reqOpts = {
     subscription: this.name
   };
 
   if (is.string(snapshot)) {
-    reqOpts.snapshot = Snapshot.formatName_(this.parent.projectId, snapshot);
+    reqOpts.snapshot = Snapshot.formatName_(this.pubsub.projectId, snapshot);
   } else if (is.date(snapshot)) {
-    reqOpts.time = {
-      seconds: Math.floor(snapshot.getTime() / 1000),
-      nanos: snapshot.getMilliseconds() * 1e6
-    };
+    reqOpts.time = snapshot;
   } else {
     throw new Error('Either a snapshot name or Date is needed to seek to.');
   }
 
-  this.parent.request(protoOpts, reqOpts, callback);
+  this.request({
+    client: 'subscriberClient',
+    method: 'seek',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, callback);
 };
 
 /**
- * Modify the ack deadline for a specific message. This method is useful to
- * indicate that more time is needed to process a message by the subscriber, or
- * to make the message available for redelivery if the processing was
- * interrupted.
+ * Sets a timeout to flush any acks/nacks that have been made since the pool has
+ * closed.
  *
- * @resource [Subscriptions: modifyAckDeadline API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/modifyAckDeadline}
+ * @private
+ */
+Subscription.prototype.setFlushTimeout_ = function() {
+  if (!this.flushTimeoutHandle_) {
+    this.flushTimeoutHandle_ = setTimeout(this.flushQueues_.bind(this), 1000);
+  }
+};
+
+/**
+ * Sets a timeout to modify the ack deadlines for any unacked/unnacked messages,
+ * renewing their lease.
  *
- * @param {object} options - The configuration object.
- * @param {string|string[]} options.ackIds - The ack id(s) to change.
- * @param {number} options.seconds - Number of seconds after call is made to
- *     set the deadline of the ack.
- * @param {Function=} callback - The callback function.
+ * @private
+ */
+Subscription.prototype.setLeaseTimeout_ = function() {
+  if (this.leaseTimeoutHandle_) {
```

Comment:
I like the early return style used here, but either way, can you make the way we do it consistent between `setFlushTimeout_` and this one?